libgalois/include/galois/graphs/LS_LC_CSR_Graph.h

```diff
@@ -155,8 +155,10 @@ class LS_LC_CSR_Graph : private boost::noncopyable {
   }
 
   template <typename E = EdgeData, typename = std::enable_if<HasEdgeData>>
-  void setEdgeData(EdgeHandle handle, E data) {
-    m_edge_data[handle] = data;
+  inline void setEdgeData(EdgeHandle handle, E data) {
+    m_edge_data.lazy_emplace_l(
+        handle, [&](EdgeData& v) { v = data; },
```

Comment:
this should probably be `[&](std::pair<Key, Val>& v)` or explicitly use `map_type::value_type& v`

Reply:
`auto` to the rescue!